pkg/mcs/tso/server/server.go

```diff
@@ -60,11 +109,148 @@ func (s *Server) Close() {
 
 // GetClient returns builtin etcd client.
 func (s *Server) GetClient() *clientv3.Client {
-	return nil
+	return s.client
 }
 
 // GetHTTPClient returns builtin http client.
 func (s *Server) GetHTTPClient() *http.Client {
+	return s.httpClient
+}
+
+// AddStartCallback adds a callback in the startServer phase.
+func (s *Server) AddStartCallback(callbacks ...func()) {
+	s.startCallbacks = append(s.startCallbacks, callbacks...)
+}
+
+// GetMember returns the member.
+func (s *Server) GetMember() *member.Member {
+	return s.member
+}
+
+// AddLeaderCallback adds the callback function when the server becomes leader.
+func (s *Server) AddLeaderCallback(callbacks ...func(context.Context)) {
```

Comment:
TSO service only has the concept of primary/replica.

Reply:
Yes, @lhy1024 has the plan to optimize it or I can do it later. I can't remove it in this pr, because @lhy1024 added this method to basic server interface and used it for resource manager. The server struct has to implement this method otherwise I need to remove it from the basic server and modify resource manager code accordingly.

Reply:
removed the leaderCallbacks member defined in the Server struct and leave AddLeaderCallback() be an empty function. The Server has to implement it as described in the previous message. We can add them back when we integrate it with the global TSO allocator in the future.

Reply:
I saw @lhy1024's pr #5994 has changed to this to AddPrimaryCallback(). My pr can integrate with his pr after he merges #5994. TSO has no primary in the cluster level, so I'll still remove leaderCallbacks member from the Server struct and leave empty AddLeaderCallback() and empty AddPrimaryCallback() later after integrating with #5994.